aws-android-sdk-core/src/test/java/com/amazonaws/http/UrlHttpClientTest.java

```diff
@@ -143,6 +157,36 @@ public void testCurlBuilderWithData() throws URISyntaxException, IOException {
         */
     }
 
+    @Test
+    public void testClientProxySettings() throws URISyntaxException, IOException {
+        conf.setCurlLogging(true);
+        final Map<String, String> headers = new HashMap<String, String>();
+        headers.put("key1", "value1");
+        headers.put("key2", "value2");
+
+        final HashSet<String> expectedCurlHeaders = new HashSet<String>();
+        for (final Map.Entry<String, String> entry : headers.entrySet()) {
+            expectedCurlHeaders.add("\"" + entry.getKey() + ":" + entry.getValue() + "\"");
+        }
+
+        final HttpRequest request = new HttpRequest("POST", new URI("https://www.test.com"),
+                headers,
+                null /* stream */);
+
+        conf.setProtocol(Protocol.HTTP);
+        conf.setProxyHost(server.getListenAddress().getHostName());
+        conf.setProxyPort(server.getListenAddress().getPort());
+
+        try {
+            client.execute(request);
+        } catch (final UnknownHostException exception) {
+            return;
+        }
+        assertNotNull(client.proxy);
```

Comment:
Can the test make the proxy return an expected value and verify it? This doesn't seem like sufficient validation.